packages/expo-modules-core/ios/FileSystemUtilities/FileSystemLegacyUtilities.swift

```diff
@@ -0,0 +1,111 @@
+// Copyright 2023-present 650 Industries. All rights reserved.
+
+import Foundation
+
+@objc(EXFileSystemLegacyUtilities)
+public class FileSystemLegacyUtilities: NSObject, EXInternalModule, EXFileSystemInterface, EXFilePermissionModuleInterface {
+  @objc
+  public var documentDirectory: String
+
+  @objc
+  public var cachesDirectory: String
+
+  var isScoped: Bool = false
+
+  @objc
+  public init(documentDirectory: String, cachesDirectory: String) {
+    self.documentDirectory = documentDirectory
+    self.cachesDirectory = cachesDirectory
+    self.isScoped = true
+
+    super.init()
+    ensureDirExists(withPath: self.cachesDirectory)
```

Comment:
@tsapeta now we ensure these dirs exist on init – this is moved over from the ExFileSystem.m module.